jraft-core/src/main/java/com/alipay/sofa/jraft/util/ThreadId.java

```diff
@@ -157,17 +157,19 @@ public void setError(final int errorCode) {
         if (this.destroyed) {
             return;
         }
-        if (this.lock.tryLock()) {
-            if (this.destroyed) {
-                this.lock.unlock();
-                return;
-            }
-            if (this.onError != null) {
-                // The lock will be unlocked in onError.
-                this.onError.onError(this, this.data, errorCode);
+        synchronized (pendingErrors) {
+            if (this.lock.tryLock()) {
+                if (this.destroyed) {
+                    this.lock.unlock();
+                    return;
+                }
+                if (this.onError != null) {
+                    // The lock will be unlocked in onError.
+                    this.onError.onError(this, this.data, errorCode);
+                }
+            } else {
+                this.pendingErrors.add(errorCode);
```

Comment:
```
private final List<Integer>    pendingErrors       = Collections.synchronizedList(new ArrayList<>());
```
Should we redeclare `pendingErrors` as An ArrayList?

Reply:
done